ndcube/mixins/ndslicing.py

```diff
@@ -40,22 +42,30 @@ def _slice(self, item):
         to add support for ``missing_axes`` and ``extra_coords`` and overwrites
         the astropy handling of wcs slicing.
         """
+
+        item = tuple(sanitize_slices(item, len(self.dimensions)))
         kwargs = super()._slice(item)
 
-        wcs, missing_axes = self._slice_wcs_missing_axes(item)
-        kwargs['wcs'] = wcs
-        kwargs['missing_axes'] = missing_axes
-        kwargs['extra_coords'] = self._slice_extra_coords(item, missing_axes)
+        # Store the original dimension of the wcs object before slicing
+        prev_dim = self.high_level_wcs.low_level_wcs.world_n_dim
 
+        # Sanitize the input arguments
+        wcs = self._slice_wcs(item)
```

Comment:
The ```item``` should be reversed here.

Reply:
it looks like it shouldn't, and that `SlicedLowLevelWCS` takes numpy ordered slices...